### PR TITLE
Add example folder (with examples in it) 

### PR DESCRIPTION
**Pull request by sylviamoss**
This adds a complete HCL2 template example that is compatible with `packer init`. Besides that, I tried making the scaffolding builder more complete with didactic comments.

- [x] Make the example release of v0.1.0 
- [x] Write README

Part of https://github.com/hashicorp/packer-plugin-scaffolding/issues/16

```shell
➜  example git:(set_example) packer init .
Installed plugin github.com/hashicorp/scaffolding v0.1.0 in "/Users/sylviamoss/.packer.d/plugins/github.com/hashicorp/scaffolding/packer-plugin-scaffolding_v0.1.0_x5.0_darwin_amd64"

```

### CODE REVIEW

File: builder/scaffolding/step_say_config.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package scaffolding
```
Comment (SwampDragons) [inline builder/scaffolding/step_say_config.go:1]: Great addition, having an actual step definition.

File: builder/scaffolding/step_say_config.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		return multistep.ActionHalt
+	}
+
+	ui.Say(fmt.Sprintf("The mock config is set to %q", s.MockConfig))
+	// Determines that should continue to the next step
+	return multistep.ActionContinue
+}
+
+// Cleanup can be used to clean up any artifact created by the step.
+// A step's clean up always run at the end of a build.
```
Comment (SwampDragons) [inline builder/scaffolding/step_say_config.go:35]:
```suggestion
// A step's clean up always run at the end of a build, regardless of whether provisioning succeeds or fails.
```

**Review by SwampDragons — APPROVED** [review]
Awesome improvements. One comment nitpick.

File: example/build.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,40 @@
+packer {
+  required_plugins {
+    scaffolding = {
+      version = ">=v0.1.0"
+      source = "github.com/hashicorp/scaffolding"
```
Comment (ksatirli) [inline example/build.pkr.hcl:5]:
```suggestion
      source  = "github.com/hashicorp/scaffolding"
```

**Pull request merged**